src/transformStateWithClones.js

```diff
@@ -65,7 +67,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const interimStates = JSON.parse(JSON.stringify(state));
+  const statesArray = [];
+
+  for (const property of transforms) {
+    if (property['operation'] === 'addProperties') {
+      for (const [key, value] of Object.entries(property['properties'])) {
+        interimStates[key] = value;
+      }
+      statesArray.push(JSON.parse(JSON.stringify(interimStates)));
+    }
+
+    if (property['operation'] === 'clear') {
+      for (const key in interimStates) {
+        delete interimStates[key];
+      }
+      statesArray.push(JSON.parse(JSON.stringify(interimStates)));
+    }
+
+    if (property.operation === 'removeProperties') {
+      for (const keys of property['properties']) {
+        delete interimStates[keys];
+      }
+      statesArray.push(JSON.parse(JSON.stringify(interimStates)));
```

Comment:
You repeated this block of code three times.
Better to write it one time after the switch loop.